### PR TITLE
Update tidb-binlog-kafka.md

### PR DESCRIPTION
**Pull request by yanyanqing**

### CODE REVIEW

File: tools/tidb-binlog-kafka.md [hunk 1]
```diff
@@ -73,10 +73,10 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
 
     为了保证数据的完整性，在 pump 运行 10 分钟左右后按顺序进行下面的操作
 
-    *  使用 generate_binlog_position 工具生成 drainer 启动需要的 savepint 文件，工具在项目 [tidb-tools](https://github.com/pingcap/tidb-tools) 中，make generate_binlog_position 编译该工具，具体的使用参考工具的 README 说明。
+    *  使用 generate_binlog_position 工具生成 drainer 启动需要的 savepoint 文件，工具在项目 [tidb-tools](https://github.com/pingcap/tidb-tools) 中，make generate_binlog_position 编译该工具，具体的使用参考工具的 README 说明。
     *  全量备份，例如 mydumper 备份 tidb
     *  全量导入备份到目标系统
-    *  设置 savepoint 文件路径，然后启动 drainer， `bin/drainer --config=conf/drainer.toml --data-dir=${drainer_savepoint_dir}`
+    *  kafka 版 drainer 的 savepoint 默认保存在下游 database tidb_binlog 下的 checkpoint 表中，如果 checkpoint 表中没有有效的数据，可以通过 savepoint 文件获取，然后启动 drainer， `bin/drainer --config=conf/drainer.toml --initial-commit-ts=${drainer_savepoint_dir/commitTS}`
```
Comment (GregoryIan) [inline tools/tidb-binlog-kafka.md]: 可以通过 savepoint 获取是什么意思？

File: tools/tidb-binlog-kafka.md [hunk 2]
```diff
@@ -73,14 +73,16 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
 
     为了保证数据的完整性，在 pump 运行 10 分钟左右后按顺序进行下面的操作
 
-    *  使用 generate_binlog_position 工具生成 drainer 启动需要的 savepint 文件，工具在项目 [tidb-tools](https://github.com/pingcap/tidb-tools) 中，make generate_binlog_position 编译该工具，具体的使用参考工具的 README 说明。
+    *  使用 generate_binlog_position 工具生成 drainer 启动需要的 savepoint 文件，工具在项目 [tidb-tools](https://github.com/pingcap/tidb-tools) 中，make generate_binlog_position 编译该工具，具体的使用参考工具的 README 说明。
     *  全量备份，例如 mydumper 备份 tidb
     *  全量导入备份到目标系统
-    *  设置 savepoint 文件路径，然后启动 drainer， `bin/drainer --config=conf/drainer.toml --data-dir=${drainer_savepoint_dir}`
+    *  kafka 版本 drainer 启动的 savepoint 默认保存在下游 database tidb_binlog 下的 checkpoint 表中，
+    如果 checkpoint 表中没有效的数据，可以通过 savepoint 文件获取，然后启动 drainer，
```
Comment (GregoryIan) [inline tools/tidb-binlog-kafka.md]: savepoint 文件在哪？

**Comment by GregoryIan** [comment]
@sunhao2017 @buggithubs  PTAL

File: tools/tidb-binlog-kafka.md [hunk 3]
```diff
@@ -73,14 +73,14 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
 
     为了保证数据的完整性，在 pump 运行 10 分钟左右后按顺序进行下面的操作
 
-    *  使用 generate_binlog_position 工具生成 drainer 启动需要的 savepint 文件，工具在项目 [tidb-tools](https://github.com/pingcap/tidb-tools) 中，make generate_binlog_position 编译该工具，具体的使用参考工具的 README 说明。
+    *  使用 generate_binlog_position 工具生成 drainer 启动需要的 savepoint 文件，工具在项目 [tidb-tools](https://github.com/pingcap/tidb-tools) 中，make generate_binlog_position 编译该工具，具体的使用参考工具的 README 说明。
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: drainer -> Drainer

File: tools/tidb-binlog-kafka.md [hunk 4]
```diff
@@ -73,14 +73,14 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
...
     *  全量备份，例如 mydumper 备份 tidb
     *  全量导入备份到目标系统
-    *  设置 savepoint 文件路径，然后启动 drainer， `bin/drainer --config=conf/drainer.toml --data-dir=${drainer_savepoint_dir}`
+    *  kafka 版本 drainer 启动的 savepoint 默认保存在下游 database tidb_binlog 下的 checkpoint 表中，如果 checkpoint 表中没有效的数据，可以通过设置 `initial-commit-ts` 启动 drainer 从指定位置开始消费 - `bin/drainer --config=conf/drainer.toml --initial-commit-ts=${commitTS}`
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: - kafka -> Kafka
- drainer -> Drainer

File: tools/tidb-binlog-kafka.md [hunk 5]
```diff
@@ -89,9 +89,23 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
     dir = "/path/pb-dir"
     ```
    
-*   Kafka 和 Zookeeper 集群的安装和配置不在本文赘述，需要在部署 TiDB-Binlog 之前部署好。Kafka 需要0.9及以上版本，且保证设置参数auto.create.topics.enable=true, 推荐部署在3～5台服务器上，机器的磁盘空间和具体的业务数据量相关。
+#### Kafka 和 Zookeeper 集群需要在部署 TiDB-Binlog 之前部署好。Kafka 需要0.9及以上版本。
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: Add a whitespace before and after "0.9".

File: tools/tidb-binlog-kafka.md [hunk 6]
```diff
@@ -89,9 +89,23 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
...
+#### kafka 集群配置推荐 
 
+|名字|数量|内存|CPU|硬盘|
+|:---:|:---:|:---:|:---:|:---:|
+|Kafka|3+|16G|8+|2+ 1TB|
+|Zookeeper|3+|8G|4+|2+ 300G|
 
+#### kafka 配置参数推荐
+    
+ - auto.create.topics.enable=true 如果还没有创建topic，kafka会在broker上自动创建topic
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: Add a whitespace between the Chinese and English characters. Please modify similar items involved in this PR.

File: tools/tidb-binlog-kafka.md [hunk 7]
```diff
@@ -89,9 +89,23 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
...
+ - broker.id 必备参数用来标识 kafka 集群，不能重复，如 broker.id = 1
+ - fs.file-max = 1000000 kafka会使用大量文件和网络 socket,建议修改成 1000000, 修改方法（vi /etc/sysctl.conf）
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: - Add whitespace
- Change the two commas to full-width (Chinese) style "，"

File: tools/tidb-binlog-kafka.md [hunk 8]
```diff
@@ -107,206 +121,212 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
   ``` 
 
 #### 使用 Binary 部署 PUMP
-
-1. PUMP 命令行参数说明
-
+    使用样例：
+    假设我们有三个 PD，三个 zookeeper，一个 TiDB, 各个节点信息如下
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: - Ditto. Comma.
- Add a colon "：" after "如下"

File: tools/tidb-binlog-kafka.md [hunk 9]
```diff
@@ -107,206 +121,212 @@ Kafka 集群用来存储由 Pump 写入的 binlog 数据，并提供给 Drainer
...
+    PD1="192.168.0.16"
+    PD2="192.168.0.15"
+    PD3="192.168.0.14"
+    ZK1="192.168.0.13"
+    ZK2="192.168.0.12"
+    ZK3="192.168.0.11"
     ```
+    在 ip="192.168.0.10" 的机器上面部署 drainer pump
+    对应的 pd 集群的 ip="192.168.0.16,192.168.0.15,192.168.0.14" 
+    对应的 kafka 集群的 zookeeper 的 ip="192.168.0.13,192.168.0.12,192.168.0.11" 以此为例，说明 pump drainer 的使用
```
Comment (lilin90) [inline tools/tidb-binlog-kafka.md]: Capitalize the first letter of the component names:

Drainer, Pump, PD, Kafka, Zookeeper

Pls update similar items involved in this PR.

**Pull request merged**